src/power_grid_model/_core/power_grid_meta.py

```diff
@@ -178,7 +179,9 @@ def initialize_array(
     component_type: ComponentTypeLike,
     shape: tuple | int,
     empty: bool = False,
-) -> SingleArray | DenseBatchArray:
+    *,
+    attributes: Iterable[str] | None = None,
```

Comment:
do we also want to support `everything` (genuine question, feel free to close it if not)?